File: speedtest01.java
public class speedtest01 {       
    public static final int END_LOOP = 1_000_000_000; // 1 Mrd.
    public static void main(String[] args) {
        // Loop from 1 to 
        int last_number = 0;
        for (int i = 0; i < END_LOOP; i++) {
            last_number++;
        }
        System.out.println("Java:last_number=" + last_number + "\n");
    }
}

File: speedtest01.pas
// compile with:
// fpc speedtest01.pas -o"speedtest01_pas.bin"

program Speedtest01;

const
    END_LOOP : UInt32 = 1000000000; // 1 Mrd.
var
    last_number : UInt32 = 0;
    i : UInt32;
begin
    for i := 0 to (END_LOOP-1) do
        last_number := last_number+1;

    WriteLn('Pascal:last_number=', last_number);
    WriteLn('--');
end.